### PR TITLE
Update specification for enhanced parts with extension resolution

### PR DESCRIPTION
**Pull request by lrhn**
Adds rules for resolving implicit extension applications in the new, more complicated, scope.

- Thanks for your contribution! Please replace this text with a description of what this PR is changing or adding and why, list any relevant issues, and review the contribution guidelines below.

### CODE REVIEW

File: working/augmentation-libraries/parts_with_imports.md [hunk 1]
```diff
@@ -296,14 +298,56 @@ library, so the library author should fix the conflict by renaming the prefix.
...
+An extension declaration being *available* has been defined as being
+declared or *imported* by the current library.
+Being imported by a library means that the library has at least one 
+import directive which *imports the extensions*, which again means that it
+imports a library which has the extension declaration in its export scope,
+and the import directive does not have a `show` or `hide` combinator 
+which hides the name of the extension declaration.
+
+With this feature, imports are not global to the entire library,
+and neither are extension availability.
```
Comment (jakemac53) [inline working/augmentation-libraries/parts_with_imports.md:318]:
```suggestion
and neither is extension availability. 
```

File: working/augmentation-libraries/parts_with_imports.md [hunk 2]
```diff
@@ -296,14 +298,56 @@ library, so the library author should fix the conflict by renaming the prefix.
...
+
+Extension availability is defined *per file*, and an extension 
+is available *in a Dart file* if any of:
+* The extension is declared by the library of the Dart file.
+* The extension is available *by import* in the Dart file.
+
+where an extension is available by import in a Dart file if any of:
+* That file contains an import directive which *imports the extension*
+* That file is a part file and the extension is available by import
+  in its parent file.
```
Comment (jakemac53) [inline working/augmentation-libraries/parts_with_imports.md:328]: This applies transitively which should probably be called out in some way

File: working/augmentation-libraries/parts_with_imports.md [hunk 3]
```diff
@@ -610,13 +655,29 @@ will already be compatible.
 
 [string_part_of_lint]: https://dart.dev/tools/linter-rules/use_string_in_part_of_directives	"use_string_in_part_of_directives lint"
 
+### Development
+
+The experiment name for this feature is `enhanced-parts`.
+
+The macro feature requires both this feature and the augmentations feature.
+Tools can choose to enable these feature automatically when the macros feature
```
Comment (jakemac53) [inline working/augmentation-libraries/parts_with_imports.md:663]:
```suggestion
Tools can choose to enable these features automatically when the macros feature
```

File: working/augmentation-libraries/parts_with_imports.md [hunk 4]
```diff
@@ -610,13 +655,29 @@ will already be compatible.
...
+is enabled, or they can enable it selectively only for code generated by macros.
+
+The augmentations feature does not require enhanced parts, it can work
+with within the existing part requirements.
```
Comment (jakemac53) [inline working/augmentation-libraries/parts_with_imports.md:667]:
```suggestion
within the existing part requirements.
```

**Review by jakemac53 — APPROVED** [review]

File: working/augmentation-libraries/parts_with_imports.md (hunk 2 above)
Comment (lrhn) [reply working/augmentation-libraries/parts_with_imports.md]: Now says "(recursively)".

**Pull request merged**